nemo_rl/distributed/model_utils.py

```
@@ -1054,3 +1060,340 @@ def backward(
+            torch.distributed.all_gather(
+                all_hidden_states, tensor_parallel_hidden_states, group=tp_group
+            )
+            tensor_parallel_hidden_states = torch.cat(all_hidden_states, dim=0)
+        chunk_size = ctx.chunk_size
+        tp_group = ctx.tp_group
+        # this is the vocab size for this partition when the output_layer is a ColumnParallelLinear
+        partition_vocab_size = output_weight_layer.size(0)
+        seq_size = int(tensor_parallel_hidden_states.shape[0])
+        num_chunks = (seq_size + chunk_size - 1) // chunk_size
+        all_grad_input_hidden_states = []
+        all_grad_input_output_layer = []
```

Ruff (0.15.2)

[error] Local variable `all_grad_input_output_layer` is assigned to but never used

Remove assignment to unused variable `all_grad_input_output_layer`

(F841)